### ISSUE TITLE
auto-restart feature

### ISSUE THREAD

**Issue by atfrase** [itzg/docker-minecraft-server#1918]
### Enhancement Type

A completely new feature

### Describe the enhancement

It would be great to have some options for the server to automatically restart itself under configurable conditions: i.e. no users connected in the last X minutes (similar to the autopause condition), only inside a certain time window, etc. When combined with SPIGET_RESOURCES this would also facilitate easier automatic updating of highly version-sensitive plugins such as Geyser, ViaVersion, or ViaBackwards.

**Comment by itzg** [comment]
The requested feature is very difficult to implement fully within the scope of the container. The auto-stop feature 

https://github.com/itzg/docker-minecraft-server#autostop

was implemented as a compromise assuming the container had been declared with an appropriate restart policy.

**Comment by DarkMagicSource** [comment]
Yeah, lots of that intricate stuff is hard to get from at the container level, ie number of players online. A while ago I was using [this](https://www.spigotmc.org/resources/1-8-1-19-3-ultimateautorestart-need-an-autorestart-plugin-grab-the-best-one-today.64414/) plugin on Paper and just had my container configured to restart `unless-stopped` and didn't provide a restart script to the plugin and it worked perfectly.

**Comment by itzg** [comment]
Yeah, "very difficult" == still quite doable 😀 I could see https://github.com/itzg/mc-server-runner/issues/19 being a part of what could feed into this logic.
